### PR TITLE
Update candidate

### PR DESCRIPTION
**Pull request by voidsong-dragonfly**
Add in comparator support for hatches & Update hatch fluid handling code
Refactors from Bryce92

### CODE REVIEW

File: src/main/java/zmaster587/libVulpes/api/material/MaterialRegistry.java
Comment (zmaster587) [inline src/main/java/zmaster587/libVulpes/api/material/MaterialRegistry.java]:
```suggestion
	 * @return an ItemStack of stackSize amount containing the product with the given material, or ItemStack.EMPTY if one does not exist
```

File: src/main/java/zmaster587/libVulpes/client/util/ProgressBarImage.java
Comment (zmaster587) [inline src/main/java/zmaster587/libVulpes/client/util/ProgressBarImage.java]:
```suggestion
        vertexbuffer.pos(x, y + height, zLevel).tex((float)(textureX) * 0.00390625F, (float)(textureY + height) * 0.00390625F).endVertex();
```

File: src/main/java/zmaster587/libVulpes/client/util/ProgressBarImage.java
Comment (zmaster587) [inline src/main/java/zmaster587/libVulpes/client/util/ProgressBarImage.java]:
```suggestion
        vertexbuffer.pos(x + width, y, zLevel).tex((float)(textureX + width) * 0.00390625F, (float)(textureY) * 0.00390625F).endVertex();
```

File: src/main/java/zmaster587/libVulpes/client/util/ProgressBarImage.java
Comment (zmaster587) [inline src/main/java/zmaster587/libVulpes/client/util/ProgressBarImage.java]:
```suggestion
        vertexbuffer.pos(x, y, zLevel).tex((float)(textureX) * 0.00390625F, (float)(textureY) * 0.00390625F).endVertex();
```

File: src/main/java/zmaster587/libVulpes/inventory/ContainerModular.java
Comment (zmaster587) [inline src/main/java/zmaster587/libVulpes/inventory/ContainerModular.java]: This whole commented out bit can probably be removed.

File: src/main/java/zmaster587/libVulpes/items/ItemLinker.java
Comment (zmaster587) [inline src/main/java/zmaster587/libVulpes/items/ItemLinker.java]: While technically correct, it's a lot less readable that using else

File: src/main/java/zmaster587/libVulpes/items/ItemMaterialBlock.java
Comment (zmaster587) [inline src/main/java/zmaster587/libVulpes/items/ItemMaterialBlock.java]: We can probably remove these commented out methods.

File: src/main/java/zmaster587/libVulpes/tile/energy/TileCreativePowerInput.java
Comment (zmaster587) [inline src/main/java/zmaster587/libVulpes/tile/energy/TileCreativePowerInput.java]: Not sure, may have been pointing to a method with a different sig at some point.

File: src/main/java/zmaster587/libVulpes/tile/multiblock/TileMultiBlock.java
Comment (zmaster587) [inline src/main/java/zmaster587/libVulpes/tile/multiblock/TileMultiBlock.java]: Should stay on multiple lines for readability, however the operator should come first, not at the end like the previous code did.

```suggestion
					if(blockBroken 
					&& globalX == destroyedPos.getX() 
					&& globalY == destroyedPos.getY() 
					&& globalZ == destroyedPos.getZ()) {
```

File: src/main/java/zmaster587/libVulpes/tile/multiblock/TileMultiBlock.java
Comment (zmaster587) [inline src/main/java/zmaster587/libVulpes/tile/multiblock/TileMultiBlock.java]: Should probably put this on multiple lines for readability
```suggestion
					if(!(structure[y][z][x] instanceof Character
					&& (Character)structure[y][z][x] == 'c')
					&& !(structure[y][z][x] instanceof Block
					&& structure[y][z][x] == Blocks.AIR
					&& world.isAirBlock(globalPos))
					&& !getAllowableBlocks(structure[y][z][x]).contains(new BlockMeta(block,meta))) {
```

File: src/main/java/zmaster587/libVulpes/tile/multiblock/TileMultiBlock.java
Comment (zmaster587) [inline src/main/java/zmaster587/libVulpes/tile/multiblock/TileMultiBlock.java]:
```suggestion
						if(block.isReplaceable(world, globalPos)
						&& structure[y][z][x] instanceof Block
						&& structure[y][z][x] == Blocks.AIR)
```

File: src/main/java/zmaster587/libVulpes/items/ItemMaterialBlock.java
Comment (voidsong-dragonfly) [reply src/main/java/zmaster587/libVulpes/items/ItemMaterialBlock.java]: That entire file was useless so I simply opted to remove it

File: src/main/java/zmaster587/libVulpes/tile/energy/TileCreativePowerInput.java
Comment (voidsong-dragonfly) [reply src/main/java/zmaster587/libVulpes/tile/energy/TileCreativePowerInput.java]: Removed

**Review by zmaster587 — APPROVED** [review]

**Comment by zmaster587** [comment]
I'm going to hold off merging this in until the AR one is merged, I don't want another situation where the LibVulpes release is too new to work with the current AR release due to method changes.

**Comment by voidsong-dragonfly** [comment]
Yeah, that's a good idea

**Pull request merged**
